### PR TITLE
scripts/verify-report.sh: handle multi-line string with single quotes

### PR DESCRIPTION
**Pull request by zonggen**
This fix handles multi-line reason strings more nicely by removing use
of xargs and concatenating strings spanning multiple lines. Note that
this script will be completely replaced by the incoming report command
of chart-verifier.

Closes: https://issues.redhat.com/browse/HELM-211
Signed-off-by: Allen Bai <abai@redhat.com>

### CODE REVIEW

**Comment by zonggen** [comment]
<details>
  <summary>Test report</summary>

```yaml
apiversion: v1
kind: verify-report
metadata:
    tool:
        verifier-version: 1.1.0
        profile:
            VendorType: partner
            version: v1.0
        chart-uri: https://redhat-developer.github.io/redhat-helm-charts/charts/wildfly-1.4.0.tgz
        digest: sha256:80630f72e50faa6cd799f8bad147f834fe2e4b181ba4a8298a40e743665fba1b
        digests:
            chart: sha256:80630f72e50faa6cd799f8bad147f834fe2e4b181ba4a8298a40e743665fba1b
            package: 1c4a83c3ea0b23f7adf22210e950ddb4a658641ecb57a97ef9a19268059412d2
        lastCertifiedTimestamp: "2021-07-09T23:13:06.36487+00:00"
        certifiedOpenShiftVersions: 4.7.5
    chart:
        name: wildfly
        home: ""
        sources: []
        version: 1.4.0
        description: Build and Deploy WildFly applications on OpenShift
        keywords: []
        maintainers:
          - name: WildFly
            email: wildfly-dev@lists.jboss.org
            url: https://wildfly.org
        icon: https://design.jboss.org/wildfly/logo/final/wildfly_logomark_256px.png
        apiversion: v2
        condition: ""
        tags: ""
        appversion: "24.0"
        deprecated: false
        annotations: {}
        kubeversion: ""
        dependencies:
          - name: wildfly-common
            version: 1.3.0
            repository: file://../wildfly-common
            condition: ""
            tags: []
            enabled: false
            importvalues: []
            alias: ""
        type: application
    chart-overrides: ""
results:
  - check: v1.0/contains-values-schema
    type: Mandatory
    outcome: PASS
    reason: Values schema file exist
  - check: v1.0/has-kubeversion
    type: Mandatory
    outcome: FAIL
    reason: Kubernetes version is not specified
  - check: v1.0/images-are-certified
    type: Mandatory
    outcome: FAIL
    reason: 'Failed to certify images : Failed to get images, error running helm template
        : execution error at (wildfly/templates/buildconfig-s2i-build-artifacts.yaml:2:4):
        Git source repository URL is required'
  - check: v1.0/has-readme
    type: Mandatory
    outcome: PASS
    reason: Chart has a README
  - check: v1.0/is-helm-v3
    type: Mandatory
    outcome: PASS
    reason: API version is V2, used in Helm 3
  - check: v1.0/contains-test
    type: Mandatory
    outcome: FAIL
    reason: Chart test files do not exist
  - check: v1.0/not-contain-csi-objects
    type: Mandatory
    outcome: PASS
    reason: CSI objects do not exist
  - check: v1.0/chart-testing
    type: Mandatory
    outcome: FAIL
    reason: |-
        Error running process: executing helm with args "install wildfly-dy22xxfx2x /root/.cache/chart-verifier/https___redhat_developer_github_io_redhat_helm_charts_charts_wildfly_1_4_0_tgz/wildfly --namespace davptest --wait --values /tmp/chart-testing-636415511/values.yaml": exit status 1
        ---
        Error: execution error at (wildfly/templates/buildconfig-s2i-build-artifacts.yaml:2:4): Git source repository URL is required
  - check: v1.0/contains-values
    type: Mandatory
    outcome: PASS
    reason: Values file exist
  - check: v1.0/not-contains-crds
    type: Mandatory
    outcome: PASS
    reason: Chart does not contain CRDs
  - check: v1.0/helm-lint
    type: Mandatory
    outcome: PASS
    reason: Helm lint successful
```
</details>

<details>
  <summary>Before fix</summary>

```shell
$ ./verify-report.sh results test-report-wildfly.yaml partner

xargs: unmatched single quote; by default quotes are special to xargs unless you use the -0 option
{"passed": 7, "failed": 5, "message": ["v1.0/has-kubeversion : Kubernetes version is not specified","v1.0/contains-test : Chart test files do not exist","v1.0/chart-testing : Error running process: executing helm with args install wildfly-dy22xxfx2x /root/.cache/chart-verifier/https___redhat_developer_github_io_redhat_helm_charts_charts_wildfly_1_4_0_tgz/wildfly --namespace davptest --wait --values /tmp/chart-testing-636415511/values.yaml: exit status 1","v1.0/chart-testing : ---","v1.0/chart-testing : Error: execution error at (wildfly/templates/buildconfig-s2i-build-artifacts.yaml:2:4): Git source repository URL is required"]}
```
</details>

<details>
  <summary>After fix</summary>
  
```shell
$ ./verify-report.sh results test-report-wildfly.yaml partner | jq

{
  "passed": 7,
  "failed": 4,
  "message": [
    "v1.0/has-kubeversion : Kubernetes version is not specified",
    "v1.0/images-are-certified : 'Failed to certify images : Failed to get images, error running helm template : execution error at (wildfly/templates/buildconfig-s2i-build-artifacts.yaml:2:4): Git source repository URL is required'",
    "v1.0/contains-test : Chart test files do not exist",
    "v1.0/chart-testing : Error running process: executing helm with args 'install wildfly-dy22xxfx2x /root/.cache/chart-verifier/https___redhat_developer_github_io_redhat_helm_charts_charts_wildfly_1_4_0_tgz/wildfly --namespace davptest --wait --values /tmp/chart-testing-636415511/values.yaml': exit status 1 --- Error: execution error at (wildfly/templates/buildconfig-s2i-build-artifacts.yaml:2:4): Git source repository URL is required"
  ]
}
```
</details>

<details>
  <summary>All pass report still works as before</summary>
  
```shell
./verify-report.sh results test-report-success.yaml partner | jq

{
  "passed": 11,
  "failed": 0
}
```
</details>

File: scripts/src/chartprreview/verify-report.sh [hunk 1]
```diff
@@ -242,21 +254,21 @@ getFails() {
...
+             continue
            fi
         elif [ "$multireason" = true ]; then
-          reason=`echo $line | xargs`
-          if [[ $line == *"Image is Red Hat certified"* ]]; then
-            outcome="PASS"
-          else
-            outcome="FAIL"
-          fi
+          reason+=`printf "$line\n" | sed "s/\"/'/g"`
```
Comment (zonggen) [inline scripts/src/chartprreview/verify-report.sh:270]: The two `sed`'s here ensures that no double quotes inside double quotes, which are probamatic for json parser.

**Review by dperaza4dustbit — APPROVED** [review]
looks good, however we are not solving the mystery of why the text inside the single quote has new lines in there. Did you find root cause for that?

**Comment by mmulholla** [comment]
> looks good, however we are not solving the mystery of why the text inside the single quote has new lines in there. Did you find root cause for that?

I think the newline are added by the yaml marshalling function.

**Comment by zonggen** [comment]
Yep, as the yaml encoder I think it does not care about the newline within the single quote.

**Review by mmulholla — APPROVED** [review]
LGTM

**Pull request merged**